osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/Components/PathControlPointVisualiser.cs

```diff
@@ -103,7 +101,7 @@ private void onControlPointsChanged(object sender, NotifyCollectionChangedEventA
 
                     // If removing before the end of the path,
                     // update indices of connections after remove location
-                    if (e.OldStartingIndex + e.OldItems.Count < oldSize)
+                    if (e.OldStartingIndex + e.OldItems.Count < Pieces.Count + e.OldItems.Count)
```

Comment:
hmm, i missed the fact that `Pieces` was being removed from in between... this reads a bit strange now. subtracting `e.OldItems.Count` from both sides of the inequality is functionally equivalent, but not sure if legible.

Reply:
I guess it still makes sense without the addition, if the index where we started removing is less than the number of items remaining in the path after removal that must mean we removed a section in the middle of the path.